tensorflow_federated/python/core/impl/executors/federating_executor.py

```diff
@@ -562,6 +564,81 @@ def _check_arg_is_anonymous_tuple(self, arg):
   async def _compute_intrinsic_federated_zip_at_clients(self, arg):
     return await self._zip(arg, placement_literals.CLIENTS, all_equal=False)
 
+  @tracing.trace
+  async def _trusted_aggregator_public_key(self):
+    
+    # Generate as string for test purpose
+    # Libsodium use variant type for the key
+    tensor_type = computation_types.TensorType(tf.string)
+
+    pk_a = await executor_utils.embed_tf_scalar_constant(
+        self, tensor_type, '132343242')
+
+    # Share public key with each clients
+    pk_a_shared = await self._compute_intrinsic_federated_value_at_clients(
+        pk_a
+    )
+    return pk_a_shared
+
+  @tracing.trace
+  async def _zip_val_key(self, val, key):
+    
+    # zip values and keys EagerValues on each client using their 
+    # respective eager executor
+    val_key_zipped = []
+    for i in range(len(val)):
+      client_ex = self._target_executors[placement_literals.CLIENTS][i]
+      val_eager_val = val[i]
+      key_eager_val = key.internal_representation[i]
+
+      k_v_zip = await client_ex.create_tuple(
+        anonymous_tuple.AnonymousTuple([(None, val_eager_val), (None, key_eager_val)])
+      )
+
+      val_key_zipped.append(k_v_zip)
+
+    return val_key_zipped
+
+  
+  @tracing.trace
+  async def _encrypt_client_tensors(self, arg):
+
+    @computations.tf_computation(tf.int32, tf.string)
+    @tf.function
+    def encrypt_tensor(x, aggregator_key):
+      nonce = tf.random.uniform((), 100, 1000, tf.int32)
+      pk_c = tf.random.uniform((), 100, 1000, tf.int32)
+      sk_c = tf.random.uniform((), 100, 1000, tf.int32)
+      variant = tf.data.Dataset.from_tensor_slices([1]) 
+
+      tf.print("This tensor is encrypted:", x)
+      tf.print("Trusted aggregator public key:", aggregator_key)
+      tf.print("Client public key:", pk_c)
+      tf.print("Client secret key:", sk_c)
+      tf.print("Nonce value: ", nonce)
+      tf.print("I can work with variant", variant._variant_tensor.dtype)
+      # Note: we should also return to the trusted aggregator
+      # the nonce and client public key as tuple. The trusted 
```

Comment:
@mortendahl - looks like tf_computation could potentially work with tf.variant type, at least when defining the graph. However not yet 100% we can use tf.variant as an input but at least tf.string would work.

Reply:
don't think we need variants after all!